### PR TITLE
feat: Allow passing a config from command line

### LINKED ISSUE #252: Allow taking config file as command line argument
i.e. `napari-micromanager my_config.cfg` should auto load the config file.

### PR DESCRIPTION
**Pull request by ianhi**
Allows for

```
napari-micromanager --config path/to/config.cfg
```

closes #252

### CODE REVIEW

File: src/napari_micromanager/main_window.py [hunk 1]
```diff
@@ -62,6 +65,13 @@ def __init__(self, viewer: napari.viewer.Viewer) -> None:
         self.destroyed.connect(self._cleanup)
         atexit.register(self._cleanup)
 
+        if config is not None:
+            try:
+                self._mmc.loadSystemConfiguration(config)
+            except FileNotFoundError:
+                # don't crash if the user passed an invalid config
```
Comment (tlambert03) [inline src/napari_micromanager/main_window.py:72]: should warn though, or somehow indicate that a passed parameter was ignored

**Comment by tlambert03** [comment]
happy to get this in @ianhi  ... but there are two comments that need addressing and merge conflicts to fix

**Comment by ianhi** [comment]
> but there are two comments that need addressing and merge conflicts to fix

Yup, I'm catching up on some github things this week. hope to address this and several other things soon.

**Comment by tlambert03** [comment]
oh actually, I can reproduce it if I run `python -m pytest -v --color=yes --cov=napari_micromanager --cov-report=xml` ... (like we do on CI).  I think you need to touch up the sys.argv parsing

File: tests/test_main.py [hunk 2]
```diff
@@ -1,18 +1,31 @@
...
+    [[""], ["", str(Path(__file__).parent / "test_config.cfg")], ["", "nonexistant"]],
+)
+def test_cli_main(argv: list) -> None:
     import napari
     from napari.qt import QtViewer
 
     with patch("napari.run") as mock_run:
         with patch("qtpy.QtWidgets.QMainWindow.show") as mock_show:
-            main()
+            with patch("napari.utils.notifications.show_warning") as mock_warning:
```
Comment (ianhi) [inline tests/test_main.py:19]: It seems that this patch isn't actually patching (i.e. the warning still gets displayed) and it is never called.

Any idea what I'm doing wrong here?

**Comment by ianhi** [comment]
> oh actually, I can reproduce it if I run python -m pytest -v --color=yes --cov=napari_micromanager --cov-report=xml ... (like we do on CI). I think you need to touch up the sys.argv parsing (or modify the test)

This was very helpful thank you!

File: tests/test_main.py (hunk 2 above)
Comment (tlambert03) [reply tests/test_main.py:19]: not immediately, but you know what... just use `warnings.warn()`, napari will intercept that anyway and show a warning in the UI (or at least it used to?)

Comment (ianhi) [reply tests/test_main.py:19]: hmm it doesn't seem to capture it in the same way. maybe i need to change the level

Comment (tlambert03) [reply tests/test_main.py:19]: oh, it's possible that the napari warnings hook hasn't yet been invoked by this point?

Comment (ianhi) [reply tests/test_main.py:19]: actually scratch that, now the show warnign is also not showing, so maybe just going away to quickly. it still shows up in teh terminal

**Comment by tlambert03** [comment]
i finished this up

**Pull request merged**